src/tools/crossgen2/ILCompiler.DependencyAnalysisFramework/DependencyAnalyzer.cs

```diff
@@ -250,7 +250,7 @@ private void ProcessMarkStack()
 
         public override void ComputeMarkedNodes()
         {
-            PerfEventSource.Log.GraphProcessingStart();
+            PerfEventSource.Log.GraphProcessingPhaseStart();
             if (_markingCompleted)
                 return;
```

Comment:
Is this return statement going to end up emitting a GraphProcessingPhaseStart without a matching GraphProcessingPhaseStop? Would it make sense to consider using "using" blocks to ensure the matching pair always gets generated by constructing a disposable helper class that would call the appropriate "Stop" in its Dispose method?

Reply:
I had it this way for finer granularity with timing (I wasn't sure if the start/stop would always be in the same function). But I agree the using is cleaner (except maybe wrt indentation hell) and safer. Unless you feel that granularity is more important, I'd be happy to change it.